### DISCUSSION TITLE
New rate limit?

### DISCUSSION THREAD

**Discussion by FerasK0** [robertodoering/harpy#655] [Q&A] [answered]
Is there a rate limit for tweets?

I was testing threads to make sure everything is functional and suddenly the timeline went blank with a 3:30 min waiting time.

![Screenshot_20220518-082224_harpy.png](https://user-images.githubusercontent.com/83197825/168963340-2cf5fd31-bb43-4191-8f17-16cec9f8e7ad.png)

**Comment by robertodoering** [comment, marked as answer]
Requesting timelines have always had a limit of 15 requests / 15 minutes

**Reply by FerasK0** [reply]
Okay, tbh i was refreshing a lot so fair enough :p

**Comment by dnestico** [comment]
So does this mean increased rate limit or more harpy will be able to fetch more tweets on the home timeline now with the V2 API?

Just wondering.

https://twittercommunity.com/t/reverse-chronological-home-timeline-in-the-twitter-api-v2/171549

<img width="692" alt="Screen Shot 2022-05-20 at 1 37 43 PM" src="https://user-images.githubusercontent.com/3034555/169582861-6434e867-4eb3-43e0-b7a0-822e52433c39.png">

**Reply by robertodoering** [reply]
Nice! With this we can finally migrate the home timeline to the new api :) very exciting

> This endpoint has a per-user rate limit of 180 requests per 15-minute window. You can retrieve the most recent 3200 Tweets.

Compared to the limit right now, which is 15 requests per 15-minute window (+ last 800 tweets only), this is so much better!

**Reply by robertodoering** [reply]
Some features are still missing from the new api, like DMs, but with the home timeline now available we can get started with the migration :)

**Reply by dnestico** [reply]
Awesome! 

And damn I thought DMs would be a thing by now but at least you can start the migration.
